classes/cliTool/UpgradeTool.php

```diff
@@ -209,6 +211,21 @@ public function checkVersion($versionInfo, $displayInfo = false)
         return $compare1;
     }
 
+    /**
+     * Check the existence of `app_key` variable in config file and print warning message if not found
+     */
+    public function checkForAppKey(): void
+    {
+        // if the app key variable `app_key` set in the config, nothing to do
+        if (PKPAppKey::hasKeyVariable()) {
+            return;
+        }
+        
+        printf("\n\e[;43mWARNING: It is noticed that there is not `app_key` variable defined in the `general` section of the config file which is necessary to cookie and other encryption purpose.\nWe suggest add the following line in the `general` section of config file.\e[0m\n\n");
+        
+        printf("\e[;44mapp_key = \e[0m\n");
+    }
```

Comment:
on `upgrade check` if 
1. `app_key` variable not defined in the config file, just returning but we can add another warning here . 
2. if app key not set, showing a warning . 

if we decided to make the app key a must, we can turn it into error/exception .